tests/client/test_advanced_search.py

```diff
@@ -3,18 +3,126 @@
+                        "neutral_citation": "",
+                        "specific_keyword": "",
+                        "order": "",
+                        "from": "",
+                        "to": "",
+                        "show_unpublished": "false",
+                        "only_unpublished": "false",
+                        "collections": "",
+                    }
+                ),
+            )
+
+            assert response == self.client.invoke.return_value
+
+    def test_invoke_called_with_all_params_when_all_parameters_provided(self):
+        """
+        Scenario: Searching with all parameters
+        Given a client instance
+        When the advanced_search method is called with all available parameters
+        Then it should call the MarkLogic module with all the parameters
+            and return the response
+        """
+        with patch.object(self.client, "invoke"):
+            response = self.client.advanced_search(
+                q="test query",
+                court="court",
+                judge="judge",
+                party="party",
+                neutral_citation="citation",
+                specific_keyword="keyword",
+                order="order",
+                date_from="2010-01-01",
+                date_to="2010-12-31",
+                page=2,
+                page_size=10,
+                show_unpublished=False,
+                only_unpublished=False,
+                collections="foo,bar",
+            )
+
+            self.client.invoke.assert_called_with(
+                "/judgments/search/search-v2.xqy",
+                json.dumps(
+                    {
+                        "court": ["court"],
+                        "judge": "judge",
+                        "page": 2,
+                        "page-size": 10,
+                        "q": "test query",
+                        "party": "party",
+                        "neutral_citation": "citation",
+                        "specific_keyword": "keyword",
+                        "order": "order",
+                        "from": "2010-01-01",
+                        "to": "2010-12-31",
+                        "show_unpublished": "false",
+                        "only_unpublished": "false",
+                        "collections": "foo,bar",
+                    }
+                ),
+            )
+
+            assert response == self.client.invoke.return_value
+
+    def test_error_raised_when_invoke_raises_an_error(self):
+        """
+        Scenario: MarkLogic error while searching
+        Given a client instance
+        When the advanced_search method is called and MarkLogic returns an error
+        Then it should raise an exception with the error message
+        """
+        error_message = "Error message from MarkLogic"
+        with patch.object(self.client, "invoke"):
+            self.client.invoke.side_effect = Exception(error_message)
+            with pytest.raises(Exception) as e:
```

Comment:
Can we say anything useful about the return type of the test? (It's the same / it's not the same)

Consider using RuntimeError rather than the more generic Exception group?

Reply:
Any exception that invoke raises will be raised in this function as there is no try catch. I have updated the test so that it asserts that the same exception object is raised though!